File: Java/프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        List<Integer> answer = new ArrayList<>();
        for (int i = 0; i < numbers.length; i++) {
            for (int j = 0; j < numbers.length; j++) {
                if (i != j) {
                    int sum = numbers[i] + numbers[j];
                    if (!answer.contains(sum)) {
                        answer.add(sum);
                    }
                }
            }
        }
        answer.sort(Integer::compareTo);
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: Kotlin/프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.kt
class Solution {
    val answer = mutableListOf<Int>()
    fun solution(numbers: IntArray): IntArray {
        dfs(0, 0, 0, numbers)
        answer.sort()
        return answer.toIntArray()
    }

    private fun dfs(idx: Int, depth: Int, sum: Int, numbers: IntArray) {
        if (depth == 2) {
            if (answer.contains(sum)) return
            answer.add(sum)
            return
        }

        for (i in idx until numbers.size) {
            dfs(i + 1, depth + 1, sum + numbers[i], numbers)
        }
    }
}